### PR TITLE
make the default repo for corebuilder work

### PR DESCRIPTION
**Pull request by whyrusleeping**
I was trying to build an app using an ipfs node, and the NodeBuilder panics if you dont specify a repo because the default repo setting doesnt provide a peerID and private key. So i added that generation logic into this.

### CODE REVIEW

File: core/builder.go [hunk 1]
```diff
@@ -28,10 +32,32 @@ func NewNodeBuilder() *NodeBuilder {
 	}
 }
 
-func defaultRepo() repo.Repo {
+func defaultRepo() (repo.Repo, error) {
+	c := cfg.Config{}
+	priv, pub, err := ci.GenerateKeyPairWithReader(ci.RSA, 1024, u.NewTimeSeededRand())
```
Comment (jbenet) [inline core/builder.go]: this is not secure. why not use `crypto.Rand` ? 1024 is not very slow.

Comment (whyrusleeping) [inline core/builder.go]: yeah, i guess thats true.

**Comment by whyrusleeping** [comment]
alright, updated.

**Comment by jbenet** [comment]
@whyrusleeping travis fails to build. rebase? 
- https://travis-ci.org/ipfs/go-ipfs/jobs/65292226#L134

**Pull request merged**